### PR TITLE
add hr_department_code and hr_employee_workingshift module

### PR DESCRIPTION
**Pull request by supakornkim**
Add field code to department model

### CODE REVIEW

**Comment by kittiu** [comment]
@supakornkim thank you very much for your 1st contribution to OCA. :)

Whether or not, your proposed features will be accepted to OCA, let's the PSC (person who maintain this repo) to decide.

I will give you initial review first, which will be beneficial for any PR in the future.

* Please read the instruction here https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0 to ensure that you code quality is inline with standard.
* Use this, https://github.com/OCA/maintainer-tools/tree/master/template/module/readme, to make document about your module. You just need to manage files in readme folder, after this PR is merged, ocabot will create the final README automatically.
* Normally we will respect commit history. For the new module, need only 1 commit. If you have multiple commit now, it is ok, but to make sure to squash it into 1 commit when it is ready for review. or you can use `git commit --amend` and `git push <your branch> -f` to overwrite the commit history without adding anew.
* Please review __ manifest __.py file. Easy way is to see how others did.

Thanks!

**Comment by supakornkim** [comment]
@kittiu Thank you for your advice. I will update it accordingly.

**Comment by supakornkim** [comment]
@kittiu Now the build is error which .... this one I don't know how to fix. 
May I have your advice please.. how to process this.

Thank you so much.

**Comment by Saran440** [comment]
@supakornkim Error because bot run long time.
Maybe you should split 1 PR to 1 module.
and I agree with @kittiu that you should review first.

File: hr_department_code/views/hr_department_code_view.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+            <field name="model">hr.department</field>
+            <field name="inherit_id" ref="hr.view_department_form"/>
+            <field name="arch" type="xml">
+              <field name="name" position="before">
+                  <field name="code"/>
+              </field>
+            </field>
+    </record>
+
+    <!-- Menus -->
```
Comment (kittiu) [inline hr_department_code/views/hr_department_code_view.xml:15]: Remove this if we don't have new menu.

File: hr_employee_workingshift/models/hr_working_shift.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from odoo import fields, models
+
+
+class HRWorkingShift(models.Model):
+    _name = "hr.workingshift"
```
Comment (kittiu) [inline hr_employee_workingshift/models/hr_working_shift.py:5]: I think the model should be `hr.working.shift`

**Comment by kittiu** [comment]
> @kittiu Now the build is error which .... this one I don't know how to fix.
> May I have your advice please.. how to process this.
> 
> Thank you so much.

I think it is the problem of Travis. You push again, and it might pass.
I see people discuss in OCA mailing list. People are proposing how to fix it. I will include the fix in this project later on.

I also agree with @Saran440  that you should split module. 1 PR is for 1 addon only, especially, they don't depend on each other.

By the way, @Saran440 do you think working shift is something already exists somewhere, or it is the new thing?

**Comment by supakornkim** [comment]
@kittiu 

I see Employee->Working time can also be used for working shift. The use case of my working shift may be too specific.
By the way, I will close this PR. and create a new one with only one module.

Thank you very much for your advice krub @kittiu @Saran440

**Pull request closed without merging**